ui/imports/shared/popups/keycard/KeycardPopupDetails.qml

```diff
@@ -228,6 +229,38 @@ QtObject {
                         return true
                     }
                     break
+
+                case Constants.keycardSharedFlow.createCopyOfAKeycard:
+                    switch (root.sharedKeycardModule.currentState.stateType) {
+                    case Constants.keycardSharedState.pluginReader:
```

Comment:
It would be easier to read with standard formatting like here: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/switch#syntax

Reply:
@micieslak I agree, this was the default formatting for my QtCreator for nested `switch`.

> Do we have some documentation regarding all those states?

Actually no. The only documentation we have are figma files and keycard lib flows https://notes.status.im/7l4KvRz4Q_WstOLQK30ksg?view lacking of appropriate requirements and documentation made the development harder.

> Can it be simulated somehow?

At this moment no, cause we need to mock all possible states a keycard may be in when running any flow and also simulate responses from the keycard library depending on the keycard state. That will be done when we come to ui tests, onces the development is done.